platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheProblems.kt

```diff
@@ -157,30 +178,11 @@ class ConfigurationCacheProblems(
             return
         }
 
-        when {
-            failDueToProblems -> {
-                // TODO - always include this as a build failure;
-                //  currently it is disabled when a serialization problem happens
-                validationFailures.accept(
-                    ConfigurationCacheProblemsException(summary.causes) {
-                        summary.textForConsole(cacheActionText, htmlReportFile)
-                    }
-                )
-            }
-
-            hasTooManyProblems -> {
-                validationFailures.accept(
-                    TooManyConfigurationCacheProblemsException(summary.causes) {
-                        summary.textForConsole(cacheActionText, htmlReportFile)
-                    }
-                )
-            }
-
-            else -> {
-                val logReportAsInfo = hasNoProblems && !startParameter.alwaysLogReportLinkAsWarning
-                val log: (String) -> Unit = if (logReportAsInfo) logger::info else logger::warn
-                log(summary.textForConsole(cacheActionText, htmlReportFile))
-            }
+        val failure = reportFailure(summary, htmlReportFile)
```

Comment:
🤔 We'd still need to call `validationFailures.accept(failure)` here, right?

Reply:
Yes, fixed.